src/HospitalAPI/Controllers/Intranet/MapController.cs

```diff
@@ -44,5 +58,56 @@ public async Task<IActionResult> GetRoomsByBuilding(int id)
             return Ok(mapRooms);
         }
 
+        [HttpPost]
+        [Route("rooms")]
+        public IActionResult CreateRoom([FromBody] CreateRoomDto createRoomDto)
+        {
+            MapFloor mapFloor = _mapService.GetFloorById(createRoomDto.MapFloorId);
+
+            Room room = createRoomDto.DtoToRoom();
+            room.FloorId = mapFloor.FloorId;
+
+            room = _roomService.Create(room);
+            if (room == null) return Problem("Error: room was not created");
+
+            MapRoom mapRoom = createRoomDto.DtoToMapRoom();
+            mapRoom.RoomId = room.Id;
+
```

Comment:
consider tearing this class into 3 other controllers (mapRoomController,mapFloorController,mapBuildingController)